Generate Serialization for CFNumberRef.

sheeparegreat opened the pull request:
#### 498e87cd28aa77d357d3f63c6001380623763aa4
<pre>
Generate Serialization for CFNumberRef.
<a href="https://bugs.webkit.org/show_bug.cgi?id=264024">https://bugs.webkit.org/show_bug.cgi?id=264024</a>
<a href="https://rdar.apple.com/117779935">rdar://117779935</a>

Reviewed by Brady Eidson.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
(IPC::encodeNumberInternal):
(IPC::decodeNumberInternal):
* Source/WebKit/Shared/cf/ArgumentCodersCF.cpp:
(IPC::ArgumentCoder&lt;CFNumberRef&gt;::encode): Deleted.
(IPC::sizeForNumberType): Deleted.
(IPC::ArgumentCoder&lt;RetainPtr&lt;CFNumberRef&gt;&gt;::decode): Deleted.
* Source/WebKit/Shared/cf/ArgumentCodersCF.h:
* Source/WebKit/Shared/cf/CFTypes.serialization.in:
* Source/WebKit/Shared/cf/CoreIPCNumber.h: Added.
(WebKit::CoreIPCNumber::numberHolderForNumber):
(WebKit::CoreIPCNumber::CoreIPCNumber):
(WebKit::CoreIPCNumber::createCFNumber const):
(WebKit::CoreIPCNumber::get const):
* Source/WebKit/Shared/cf/CoreIPCNumber.serialization.in: Copied from Source/WebKit/Shared/cf/CFTypes.serialization.in.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IPCSerialization.mm:
(TEST):

Canonical link: <a href="https://commits.webkit.org/270107@main">https://commits.webkit.org/270107@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ed0d57b3d1dd31aba7316c611349107ca7294862

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24580 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/3129 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25837 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26703 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22584 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24849 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/4822 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/563 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22971 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24825 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2217 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21224 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27286 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1940 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/22152 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28337 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22450 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22485 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26133 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1839 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/162 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3148 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5890 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2294 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2210 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

beidson commented on Source/WebKit/Shared/cf/CoreIPCNumber.h:
```diff
@@ -0,0 +1,170 @@
+    }
+
+private:
+    NumberHolder m_number;
```
m_numberHolder

beidson commented on Source/WebKit/Shared/cf/CoreIPCNumber.h:
```diff
@@ -0,0 +1,170 @@
+
+    CoreIPCNumber::NumberHolder get() const
+    {
+        return m_number;
```
m_numberHolder

beidson reviewed (approved):
We should turn around and apply this to NSNumber via encodeNumberInternal
If not in this patch, then right after.

Fine as-is with some naming nit picks.

sheeparegreat committed:
Generate Serialization for CFNumberRef.
https://bugs.webkit.org/show_bug.cgi?id=264024
rdar://117779935

Reviewed by Brady Eidson.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
(IPC::encodeNumberInternal):
(IPC::decodeNumberInternal):
* Source/WebKit/Shared/cf/ArgumentCodersCF.cpp:
(IPC::ArgumentCoder<CFNumberRef>::encode): Deleted.
(IPC::sizeForNumberType): Deleted.
(IPC::ArgumentCoder<RetainPtr<CFNumberRef>>::decode): Deleted.
* Source/WebKit/Shared/cf/ArgumentCodersCF.h:
* Source/WebKit/Shared/cf/CFTypes.serialization.in:
* Source/WebKit/Shared/cf/CoreIPCNumber.h: Added.
(WebKit::CoreIPCNumber::numberHolderForNumber):
(WebKit::CoreIPCNumber::CoreIPCNumber):
(WebKit::CoreIPCNumber::createCFNumber const):
(WebKit::CoreIPCNumber::get const):
* Source/WebKit/Shared/cf/CoreIPCNumber.serialization.in: Copied from Source/WebKit/Shared/cf/CFTypes.serialization.in.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IPCSerialization.mm:
(TEST):

Canonical link: https://commits.webkit.org/270107@main
